src/api/app/models/repository.rb

```diff
@@ -123,6 +123,28 @@ def expand_all_repositories
     repositories.uniq
   end
 
+  def cycles(arch)
+    # skip all packages via package=- to speed up the api call, we only parse the cycles anyway
+    deps = BuilddepInfo.find(project: project.name, package: '-', repository: name, arch: arch)
+    cycles = deps.each(:cycle).map { |cycle| cycle.each(:package).map(&:text) }
+
+    merged_cycles = []
+    cycles.each do |cycle|
+      # We look up all other cycles that intersect with this cycle
```

Comment:
Is this comment really needed? The name of the variable makes it IMO clear enough